Mathlib/NumberTheory/FLT/MasonStothers.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2024 Jineon Back and Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jineon Baek, Seewoo Lee
+-/
+import Init.Data.Nat.Lemmas
+import Mathlib.Algebra.EuclideanDomain.Defs
+import Mathlib.Algebra.Polynomial.RingDivision
+import Mathlib.Algebra.Ring.Regular
+import Mathlib.RingTheory.Polynomial.Wronskian
+import Mathlib.RingTheory.Radical
+import Mathlib.RingTheory.UniqueFactorizationDomain
+import Mathlib.Algebra.Polynomial.FieldDivision
```

Comment:
Looks like a bit too many imports. Could you please write `#min_imports` at the bottom of the file and copy here whatever it gives you?